src/ufo.rs

```diff
@@ -163,40 +236,45 @@ impl Ufo {
             };
 
             let kerning_path = path.join(KERNING_FILE);
-            let kerning = if kerning_path.exists() {
+            let kerning = if kerning_path.exists() && self.data_request.kerning {
                 let kerning: Kerning = plist::from_file(kerning_path)?;
                 Some(kerning)
             } else {
                 None
             };
 
             let features_path = path.join(FEATURES_FILE);
-            let mut features = if features_path.exists() {
+            let mut features = if features_path.exists() && self.data_request.features {
                 let features = fs::read_to_string(features_path)?;
                 Some(features)
             } else {
                 None
             };
 
+            let layers: Vec<LayerInfo>;
             let glyph_names = NameList::default();
-            let mut contents = match meta.format_version {
-                FormatVersion::V3 => {
-                    let contents_path = path.join(LAYER_CONTENTS_FILE);
-                    let contents: Vec<(String, PathBuf)> = plist::from_file(contents_path)?;
-                    contents
-                }
-                _older => vec![(DEFAULT_LAYER_NAME.into(), DEFAULT_GLYPHS_DIRNAME.into())],
-            };
-
-            let layers: Result<Vec<LayerInfo>, Error> = contents
-                .drain(..)
-                .map(|(name, p)| {
-                    let layer_path = path.join(&p);
-                    let layer = Layer::load_impl(&layer_path, &glyph_names)?;
-                    Ok(LayerInfo { name, path: p, layer })
-                })
-                .collect();
-            let layers = layers?;
+            if self.data_request.layers {
+                let mut contents = match meta.format_version {
+                    FormatVersion::V3 => {
+                        let contents_path = path.join(LAYER_CONTENTS_FILE);
+                        let contents: Vec<(String, PathBuf)> = plist::from_file(contents_path)?;
+                        contents
+                    }
+                    _older => vec![(DEFAULT_LAYER_NAME.into(), DEFAULT_GLYPHS_DIRNAME.into())],
+                };
+
+                let layers_r: Result<Vec<LayerInfo>, Error> = contents
+                    .drain(..)
+                    .map(|(name, p)| {
+                        let layer_path = path.join(&p);
+                        let layer = Layer::load_impl(&layer_path, &glyph_names)?;
+                        Ok(LayerInfo { name, path: p, layer })
+                    })
+                    .collect();
+                layers = layers_r?;
+            } else {
+                layers = Vec::new();
```

Comment:
The idiomatic way to do this would be not to assign in each block, but to use the fact that blocks in rust are all expressions; that is, I would do:

```rust
let layers = if self.data_request.layers {
    // load layers
} else {
    Vec::new()
}
```

no big deal of course but nice to be consistent.